packages/runtime/container-runtime/src/deltaManagerSummarizerProxy.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDeltaManager, ReadOnlyInfo } from "@fluidframework/container-definitions";
+import { DeltaManagerProxyBase } from "@fluidframework/container-utils";
+import { IDocumentMessage, ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { summarizerClientType } from "./summary";
+
+/**
+ * Proxy to the real IDeltaManager for restricting certain access to layers below container runtime in summarizer clients:
+ * - Summarizer client should be read-only to layers below the container runtime to restrict local changes.
+ * - Summarizer client should not be active to layers below the container runtime to restrict local changes.
+ */
+export class DeltaManagerSummarizerProxy
+	extends DeltaManagerProxyBase
+	implements IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>
+{
+	public get active(): boolean {
+		// Summarize clients should not be active. There shouldn't be any local changes (writes) in the summarizer
+		// except for the SummarizeOp which is generated by the runtime.
+		return this.isSummarizerClient ? false : this.deltaManager.active;
+	}
+
+	public get readOnlyInfo(): ReadOnlyInfo {
+		// Summarizer clients should be read-only as far as the runtime and layers below are concerned. There shouldn't
+		// be any local changes (writes) in the summarizer except for the SummarizeOp which is generated by the runtime.
+		if (this.isSummarizerClient) {
+			return {
+				readonly: true,
```

Comment:
if the real DM also says `readonly: true`, maybe pass through the readonly info as-is.

Reply:
I am not sure about that. I think we need to have a defined readonly info for summarizer clients and use those. For exampe, if the main deltamanager is readonly because it was forced to, the summarizer shouldn't say so. The summarizer is read-only because it is a system (runtime) client and that is why you cannot write.

Reply:
Ok I like that perspective, sounds good.  If needed we can even add another field `summarizer: true | undefined` to this info object which is true here